deepchem/trans/transformers.py

```diff
@@ -682,14 +706,54 @@ def X_transform(self, X_target):
     """
     X_target2 = []
     n_features = X_target.shape[1]
-    similarity = np.matmul(X_target, np.transpose(self.X)) / (
-        n_features - np.matmul(1 - X_target, np.transpose(1 - self.X)))
+    print('start similarity calculation')
+    time1 = time.time()
+    similarity = IRVTransformer.matrix_mul(X_target, np.transpose(self.X)) / (
+        n_features - IRVTransformer.matrix_mul(1 - X_target,
+                                               np.transpose(1 - self.X)))
+    time2 = time.time()
+    print('similarity calculation takes %i s' % (time2 - time1))
     for i in range(self.n_tasks):
       X_target2.append(self.realize(similarity, self.y[:, i], self.w[:, i]))
     return np.concatenate([z for z in np.array(X_target2)], axis=1)
 
+  @staticmethod
+  def matrix_mul(X1, X2, shard_size=5000):
```

Comment:
From our offline discussion, this is to handle large matrices. Could you add a quick comment describing the strategy?